src/main/java/com/myhome/services/springdatajpa/PaymentSDJpaService.java

```diff
@@ -0,0 +1,127 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.myhome.services.springdatajpa;
+
+import com.myhome.controllers.dto.PaymentDto;
+import com.myhome.controllers.dto.mapper.PaymentMapper;
+import com.myhome.domain.CommunityAdmin;
+import com.myhome.domain.HouseMember;
+import com.myhome.domain.HouseMemberDocument;
+import com.myhome.domain.Payment;
+import com.myhome.repositories.CommunityAdminRepository;
+import com.myhome.repositories.HouseMemberDocumentRepository;
+import com.myhome.repositories.HouseMemberRepository;
+import com.myhome.repositories.PaymentRepository;
+import com.myhome.services.HouseMemberDocumentService;
+import com.myhome.services.PaymentService;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * Implements {@link PaymentService} and uses Spring Data JPA Repository to do its work
+ */
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class PaymentSDJpaService implements PaymentService {
+  private final PaymentRepository paymentRepository;
+  private final PaymentMapper paymentMapper;
+  private final CommunityAdminRepository communityAdminRepository;
+  private final HouseMemberRepository houseMemberRepository;
+  private final HouseMemberDocumentService houseMemberDocumentService;
+  private final HouseMemberDocumentRepository houseMemberDocumentRepository;
+
+  @Override
+  public PaymentDto schedulePayment(PaymentDto request) {
+    generatePaymentId(request);
+    return createPaymentInRepository(request);
+  }
+
+  @Override
+  public Set<Payment> listAll() {
+    return listAll(200, 0);
+  }
+
+  @Override
+  public Set<Payment> listAll(Integer limit, Integer start) {
+    return paymentRepository.findAll(PageRequest.of(start, limit)).toSet();
+  }
+
+  @Override
+  public Optional<PaymentDto> getPaymentDetails(String paymentId) {
+    Optional<Payment> payment = paymentRepository.findByPaymentId(paymentId);
+
+    return payment.map(paymentMapper::paymentToPaymentDto);
+  }
+
+  @Override
+  public HouseMember getHouseMember(String memberId) {
+    HouseMember member = houseMemberRepository.findByMemberId(memberId).orElse(null);
+
+    if (member == null)
+      return null;
+
+    if (member.getHouseMemberDocument() == null) {
+      Optional<HouseMemberDocument> documentOpt = houseMemberDocumentService.findHouseMemberDocument(memberId);
+      HouseMemberDocument document;
+
+      if (documentOpt.isPresent()) {
+        document = documentOpt.get();
+      } else {
+        document = new HouseMemberDocument();
+        document.setId(member.getId());
+      }
+      member.setHouseMemberDocument(document);
+    }
```

Comment:
It is necessary to add empty document to house member? Can you explain why you do this?